src/prefect/utilities/git.py

```diff
@@ -40,6 +36,8 @@ def __init__(
         self.clone_depth = clone_depth
 
     def __enter__(self) -> "TemporaryGitRepo":
+        from dulwich.porcelain import clone
```

Comment:
Exceptions during these import errors should be caught and re-raised e.g. https://github.com/PrefectHQ/prefect/blob/d40dabb9adde669f14e43b52faec2d860b267225/src/prefect/storage/github.py#L74-L79

Reply:
Good catch, added!